compiler/lowerings.nim

```diff
@@ -135,9 +135,14 @@ proc lowerTupleUnpackingForAsgn*(g: ModuleGraph; n: PNode; idgen: IdGenerator; o
 proc lowerSwap*(g: ModuleGraph; n: PNode; idgen: IdGenerator; owner: PSym): PNode =
   result = newNodeI(nkStmtList, n.info)
   # note: cannot use 'skTemp' here cause we really need the copy for the VM :-(
-  var temp = newSym(skVar, getIdent(g.cache, genPrefix), nextId(idgen), owner, n.info, owner.options)
+  let temp =
+    if owner == nil:
```

Comment:
Owner shoulnd't be nil.

Reply:
I will look into this tomorrow.

Reply:
`c.prc` != nil but `c.prc.sym` == nil

Reply:
I think `c.prc` is initialized in `refresh` proc, so it isn't nil, but `c.prc.sym` is nil.
```nim
proc refresh*(c: PCtx, module: PSym; idgen: IdGenerator) =
  c.module = module
  echo "refresh: ", module
  c.prc = PProc(blocks: @[])
  c.loopIterations = c.config.maxLoopIterationsVM
  c.idgen = idgen
```



```
.refresh: test@369098752
mSwap: test@369098752
```